.changeset/metal-sloths-join.md

```diff
@@ -0,0 +1,24 @@
+---
+"fluid-framework": minor
+"@fluidframework/tree": minor
+---
+---
+"section": tree
+---
+
+Disallow some invalid and unsafe ObjectNode field assignments at compile time
```

Comment:
This PR builds upon  https://github.com/microsoft/FluidFramework/pull/23053 . This changeset will probably need to be rewritten based on the major changes.